pyccel/parser/semantic.py

```diff
@@ -1109,11 +1133,36 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             if not arr_in_multirets:
                 self._ensure_target(rhs, d_lhs)
 
-            var = self.check_for_variable(name)
+            var = self.check_for_variable(lhs)
 
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
 
+                if isinstance(lhs, DottedName):
+                    prefix = self._visit(DottedName(*lhs.name[:-1]))
+                    class_def = prefix.cls_base
+                    if isinstance(class_def, ClassDef) and lhs.name[0] == 'self':
+                        var      = self.get_variable('self')
+
+                        # Collect the name that should be used in the generated code
+                        attribute_name = lhs.name[-1]
+                        new_name = self.scope.get_expected_name(attribute_name)
```

Comment:
Is this the correct scope? Or should it be the class scope? Are class variables correctly inserted in the symbolic stage?

Reply:
Oops I mean syntactic

Reply:
If the test is passing this is probably not critical for now

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1484/commits/b9019b1faba522c9db5185e165a5a9cbaa619b90.